publisher.go

```diff
@@ -14,17 +14,15 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
-package kubewatcher
+package fission
 
-import (
-	"k8s.io/client-go/1.5/pkg/watch"
-)
+import "net/http"
 
 type (
 	Publisher interface {
-		// Publish an event to a "target".  Target's meaning depends on the
+		// Publish an request to a "target".  Target's meaning depends on the
 		// publisher: it's a URL in the case of a webhook publisher, or a queue
 		// name in a queue-based publisher such as NATS.
-		Publish(event watch.Event, target string)
+		Publish(req *http.Request)
```

Comment:
@soamvasani 

Yes. I think of that for a moment. But we have to pass almost all the params to build a http request to the channel (body, headers, target, method), and in the service we have only the http request as the protocol. So I used the request instead.
To make things simpler, we may restrict the protocol to a subset of http request. For example, only post method with string as body and custom headers as you suggest. And then your interface is universal for all the trigger events.